source/host_resolver.c

```diff
@@ -1474,7 +1474,6 @@ static int default_resolve_host(
             res(resolver, host_name, AWS_OP_SUCCESS, &callback_address_list, user_data);
```

Comment:
found another place where the callbacks can fire, even though setup fails:
on line 1411, `create_and_init_host_entry()` sets up a `struct host_entry` and if something goes wrong at any point while setting it up, calls `s_clean_up_host_entry()`, which iterates over a list of callbacks and invokes them.

We should NOT fire those callbacks unless the `host_entry` was completely set up.

Reply:
also I see that we're never checking the return value of `aws_thread_launch()`

Reply:
Good catch, fixed.